### PR TITLE
rewrite python scripts for pylint plus improvements for custom builds

### PR DESCRIPTION
**Pull request by IanBoden**
…uild)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ kind: Pod
 spec:
   containers:
   - name: amlen-centos7-build
-    image: quay.io/amlen/amlen-builder-centos7:1.0.0.5
+    image: quay.io/amlen/amlen-builder-centos7:latest
```
Comment (jonquark) [inline Jenkinsfile]: Is this change deliberate? I thought we changed to specific versions so when we branch off e.g. 1.1 maintenance branch its not broken by dev changes?

File: server_main/scripts/run.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#
+
+SECRETS_DIR=/secrets
+DESTINATION=/tmp/userfiles/
+
+mkdir -p -m 2770 ${DESTINATION}
+
+cp /secrets/sslcerts/ca/ca_public_cert.pem ${DESTINATION}
+cp /secrets/sslcerts/device/tls.crt ${DESTINATION}
```
Comment (jonquark) [inline server_main/scripts/run.sh:20]: Am I right in thinking that if these secrets change whilst we are running (e.g. cert is renewed) we won't notice.... or is this there something elsewhere watching the files?

File: Jenkinsfile (hunk 1 above)
Comment (IanBoden) [reply Jenkinsfile]: I've now changed it to 1.0.0.6 although trying to rebuild the other images all seem to be hanging with no helpful output

File: server_main/scripts/run.sh (hunk 2 above)
Comment (IanBoden) [reply server_main/scripts/run.sh:20]: I'm thinking the readiness probe could check that if the secrets exist and are different to the ones in /var/lib/amlen/data/certificates then the probe could fail, which would result in the pod restarting which would then sort it all out. Does that seem reasonable?

Comment (jonquark) [reply server_main/scripts/run.sh:20]: Yes - that seems reasonable (or something in the container can watch the file and restart the server process but either way - lets leave it as a separate item.

**Review by jonquark — APPROVED** [review]
